core/src/main/kotlin/com/bumble/appyx/core/children/ChildNodeCreationManager.kt

```diff
@@ -105,8 +159,59 @@ internal class ChildNodeCreationManager<Routing : Any>(
         }
     }
 
+    private fun childBuildContext(savedState: SavedStateMap?): BuildContext =
+        BuildContext(
+            ancestryInfo = AncestryInfo.Child(parentNode),
+            savedStateMap = savedState,
+            customisations = customisations.getSubDirectoryOrSelf(parentNode::class),
+        )
+
+    private fun childEntry(
+        key: RoutingKey<Routing>,
+        savedState: SavedStateMap?,
+        suspended: Boolean,
+    ): ChildEntry<Routing> =
+        if (suspended) {
+            ChildEntry.Suspended(key, savedState)
+        } else {
+            ChildEntry.Initialized(
+                key = key,
+                node = parentNode
+                    .resolve(key.routing, childBuildContext(savedState))
+                    .build()
+            )
+        }
+
+    private fun ChildEntry<Routing>.initialize(): ChildEntry.Initialized<Routing> =
+        when (this) {
+            is ChildEntry.Initialized -> this
+            is ChildEntry.Suspended ->
+                ChildEntry.Initialized(
+                    key = key,
+                    node = parentNode.resolve(
+                        routing = key.routing,
+                        buildContext = childBuildContext(savedState),
+                    ).build()
+                )
+        }
+
+    @Suppress("ForbiddenComment")
+    private fun ChildEntry<Routing>.suspend(): ChildEntry.Suspended<Routing> =
+        when (this) {
+            is ChildEntry.Suspended -> this
+            is ChildEntry.Initialized ->
+                ChildEntry.Suspended(
+                    key = key,
+                    // TODO: Not able to get a scope from Compose here, providing fake one
+                    savedState = node.saveInstanceState { true },
+                )
+        }
+
     private companion object {
         const val KEY_CHILDREN_STATE = "ChildrenState"
+
+        private fun <T, R : Any> Collection<T>.mapNotNullToSet(mapper: (T) -> R?): Set<R> =
```

Comment:
inline?

Reply:
Why? JVM can inline it by itself if required.

Reply:
That assumes a user of this library is using R8. I don't mind if you don't inline. though a lot of these types of functions in the kotlin lib are inlined